### PR TITLE
Add basic spvm backend

### PR DESCRIPTION
**Pull request by asdil12**

### CODE REVIEW

**Comment by okurz** [comment]
LGTM. If the test can at least do *something* with it I would even merge it and extend later.

**Comment by asdil12** [comment]
@okurz so let's do it then.

File: backend/spvm.pm [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
```
Comment (Martchus) [inline backend/spvm.pm]: Isn't this a new file?

Comment (asdil12) [reply backend/spvm.pm]: yes, it is

Comment (asdil12) [reply backend/spvm.pm]: updated (C)

File: backend/spvm.pm [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+use testapi qw(get_var get_required_var check_var);
+
+use IO::Select;
+
+# supporting the minimal command set of NovaLink through a ssh tunnel
+
+sub new {
+    my $class = shift;
+    my $self  = $class->SUPER::new;
+    get_required_var('WORKER_HOSTNAME');
```
Comment (ldevulder) [inline backend/spvm.pm]: This variable is not used... Should not be `NOVALINK_HOSTNAME` instead?

File: backend/spvm.pm [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        'novalink-ssh',
+        'ssh-xterm',
+        {
+            hostname => get_required_var('NOVALINK_HOSTNAME'),
+            password => get_required_var('NOVALINK_PASSWORD'),
+            username => get_var('NOVALINK_USERNAME', 'root')});
+    $ssh->backend($self);
+
+    # remove backend.crashed
+    $self->unlink_crash_file;
```
Comment (ldevulder) [inline backend/spvm.pm]: Test on my lab failed with unfound `unlink_crash_file`, I searched it but was not able found it...

File: backend/spvm.pm (hunk 2 above)
Comment (coolo) [reply backend/spvm.pm]: it is used by testapi

File: backend/spvm.pm (hunk 3 above)
Comment (coolo) [reply backend/spvm.pm]: that function is dead by now - the commits have been lying around for a while

Comment (asdil12) [reply backend/spvm.pm]: So shall I remove that line?

Comment (coolo) [reply backend/spvm.pm]: yes, see #930

File: backend/spvm.pm [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+sub stop_serial_grab {
+    my ($self) = @_;
+
+    $self->stop_ssh_serial;
+    return;
+}
+
+1;
+
+# vim: set sw=4 et:
```
Comment (okurz) [inline backend/spvm.pm]: I think we have this line removed from all files already, please delete

File: backend/spvm.pm [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+use IO::Select;
+
+# supporting the minimal command set of NovaLink through a ssh tunnel
+
+sub new {
+    my $class = shift;
+    my $self  = $class->SUPER::new;
+    get_required_var('WORKER_HOSTNAME');
+
+    return $self;
```
Comment (ldevulder) [inline backend/spvm.pm]: No so important, but it could be simplify like this: `return $class->SUPER::new;` (seen in others backend).

**Comment by okurz** [comment]
LGTM

File: backend/spvm.pm (hunk 5 above)
Comment (foursixnine) [reply backend/spvm.pm]: It would end up being the same.

**Comment by foursixnine** [comment]
Aside from the package not building, it seems that I forgot to check for  backend/spvm.pm being added into the makefile @asdil12 can you take a look?

**Comment by asdil12** [comment]
@foursixnine --> https://github.com/os-autoinst/os-autoinst/pull/989

**Pull request merged**